src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -5,25 +5,44 @@ import java.util.UUID
 
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.ClusterStatus
-import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.{GoogleBucket, GoogleProject, GoogleServiceAccount}
+import org.broadinstitute.dsde.workbench.leonardo.model.LabelMap.LabelMap
 import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat}
 
-// maybe we want to get fancy later
-object ModelTypes {
-  type GoogleProject = String
-  type GoogleServiceAccount = String
-  type GoogleBucket = String
+sealed trait TypedString { val value: String }
+case class ClusterName(value: String) extends TypedString
+case class GoogleProject(value: String) extends TypedString
+case class GoogleServiceAccount(value: String) extends TypedString
+case class GoogleBucket(value: String) extends TypedString
+case class OperationName(value: String) extends TypedString
+case class IP(value: String) extends TypedString
+case class LabelKey(value: String) extends TypedString
+case class LabelValue(value: String) extends TypedString
+
+object TypedString {
+  // implicitly convert to string, but not the other direction
+  import scala.language.implicitConversions
+  implicit def value(ts: TypedString): String = ts.value
 }
 
 object ClusterStatus extends Enumeration {
   type ClusterStatus = Value
   val Unknown, Creating = Value
 }
 
+object LabelMap {
+  type LabelMap = Map[LabelKey, LabelValue]
+  def apply(m: Map[String, String]): LabelMap = {
```

Comment:
Now I'm thinking these `Label` types are overkill.  The label map *is* a generic string map, so `Map[String, String]` is appropriate.